File: app/src/main/java/id/ac/uasbungaa/ui/activity/Dashboard200020016Activity.java
package id.ac.uasbungaa.ui.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import id.ac.uasbungaa.data.local.BarangHelper;
import id.ac.uasbungaa.data.local.UserHelper;
import id.ac.uasbungaa.data.model.User;
import id.ac.uasbungaa.databinding.ActivityDashboard200020016Binding;
import id.ac.uasbungaa.helper.LocalDataHelper;
import id.ac.uasbungaa.ui.activity.barang.Barang200020016Activity;
import id.ac.uasbungaa.ui.activity.user.User200020016Activity;

import java.util.ArrayList;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class Dashboard200020016Activity extends AppCompatActivity {
    private ActivityDashboard200020016Binding binding;
    private UserHelper userHelper;
    private BarangHelper barangHelper;
    private ArrayList allUser;
    private ArrayList allFlower;
    @NotNull
    public static final String EXTRA_USER = "EXTRA_USER";

    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.binding = ActivityDashboard200020016Binding.inflate(this.getLayoutInflater());

        this.setContentView(binding.getRoot());
        ActionBar supportActionBar = this.getSupportActionBar();
        if (supportActionBar != null) {
            supportActionBar.setTitle("Dashboard");
        }

        this.userHelper = new UserHelper(this);
        this.barangHelper = new BarangHelper(this);
        User username = this.getIntent().getParcelableExtra(EXTRA_USER);

        binding.tvWelcome.setText((CharSequence) ("Welcome " + (username != null ? username.getUsername() : null)));

        binding.btnUser.setOnClickListener((OnClickListener) (new OnClickListener() {
            public final void onClick(View it) {
                handleUserClicked();
            }
        }));

        binding.btnItem.setOnClickListener((OnClickListener) (new OnClickListener() {
            public final void onClick(View it) {
                handleItemClicked();
            }
        }));
    }

    protected void onResume() {
        super.onResume();

        this.allUser = LocalDataHelper.INSTANCE.getAllUser(userHelper);
        this.allFlower = LocalDataHelper.INSTANCE.getAllBarang(barangHelper);
        binding.tvUser.setText(String.valueOf(allUser.size()));
        binding.tvItem.setText(String.valueOf(allFlower.size()));
    }

    private final void handleItemClicked() {
        Intent intent = new Intent(this, Barang200020016Activity.class);
        this.startActivity(intent);
    }

    private final void handleUserClicked() {
        Intent intent = new Intent(this, User200020016Activity.class);
        this.startActivity(intent);
    }
}

File: app/src/main/java/id/ac/uasbunga/ui/activity/Dashboard200020016Activity.kt
package id.ac.uasbunga.ui.activity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import id.ac.uasbunga.data.local.BarangHelper
import id.ac.uasbunga.data.local.UserHelper
import id.ac.uasbunga.data.model.Barang
import id.ac.uasbunga.data.model.User
import id.ac.uasbunga.databinding.ActivityDashboard200020016Binding
import id.ac.uasbunga.helper.LocalDataHelper

class Dashboard200020016Activity : AppCompatActivity() {
    companion object {
        const val EXTRA_USER = "EXTRA_USER"
    }

    private lateinit var binding: ActivityDashboard200020016Binding
    private lateinit var userHelper: UserHelper
    private lateinit var barangHelper: BarangHelper
    private lateinit var allUser: ArrayList<User>
    private lateinit var allFlower: ArrayList<Barang>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDashboard200020016Binding.inflate(layoutInflater)
        setContentView(binding.root)

        supportActionBar?.title = "Dashboard"

        userHelper = UserHelper.getInstance(this)
        barangHelper = BarangHelper.getInstance(this)

        val username = intent.getParcelableExtra<User>(EXTRA_USER)
        binding.tvWelcome.text = "Welcome ${username?.username}"

        binding.btnUser.setOnClickListener {
            handleUserClicked()
        }

        binding.btnItem.setOnClickListener {
            handleItemClicked()
        }
    }

    override fun onResume() {
        super.onResume()
        allUser = LocalDataHelper.getAllUser(userHelper)
        allFlower = LocalDataHelper.getAllBarang(barangHelper)
        binding.tvUser.text = allUser.count().toString()
        binding.tvItem.text = allFlower.count().toString()
    }

    private fun handleItemClicked() {
        val intent = Intent(this, Barang200020016Activity::class.java)
        startActivity(intent)
    }

    private fun handleUserClicked() {
        val intent = Intent(this, User200020016Activity::class.java)
        startActivity(intent)
    }
}